packages/plugins/RSS3/src/SNSAdaptor/components/FeedCard.tsx

```diff
@@ -171,45 +161,78 @@ export const FeedCard = memo(({ feed, address, onSelect }: FeedCardProps) => {
                 </Card>
             )
         }
-        if (feed.tags.includes(TAG.Token) || feed.tags.includes(TAG.ETH)) {
-            return (
-                <TokenIcon
-                    pluginID={NetworkPluginID.PLUGIN_EVM}
-                    chainId={ChainID[feed.metadata?.network ?? 'ethereum']}
-                    address={feed.metadata?.token_address ?? ZERO_ADDRESS}
-                />
-            )
-        }
-        if (feed.tags.includes(TAG.Gitcoin)) {
-            return (
-                <img
-                    className={classes.img}
-                    src={feed.attachments?.find((attachment) => attachment?.type === 'logo')?.address}
-                />
-            )
+        if (feed.tag === Tag.Donation) {
+            const action = feed.actions[0] as RSS3BaseAPI.ActionGeneric<RSS3BaseAPI.Tag.Donation>
+            const logo = action.metadata?.logo
+            return <img className={classes.img} src={logo} />
         }
         return null
     }, [feed])
 
+    const normalizedFeed: RSS3Feed = useMemo(() => {
```

Comment:
Better not to include a big chunk of data progressing in a UI component.

Reply:
Abstract as a custom hook.